l10n_it_intrastat_statement/tests/test_intrastat_statement.py

```diff
@@ -229,6 +235,43 @@ def test_statement_purchase_refund(self):
         self.assertEqual(len(file_lines), 3)
         self.assertSetEqual({len(line) for line in file_lines}, {75, 130, 119})
 
+    def test_statement_purchase_refund_no_period(self):
+        bill = self._get_intrastat_computed_bill(date="2018-12-15")
+
+        bill_refund = self._create_move_refund(bill, date="2019-01-10")
+        # This refund will be subtracted from bill
+        bill_refund.update(
+            {
+                "intrastat": True,
+            }
+        )
+        bill_refund.action_post()
+        bill_refund.compute_intrastat_lines()
+
+        statement = self.statement_model.create(
+            {
+                "period_number": bill_refund.invoice_date.month,
+                "fiscalyear": bill_refund.invoice_date.year,
+            }
+        )
+
+        # Check that before computation, file generation raises an exception
+        # because statement is still empty
+        with self.assertRaises(ValidationError):
+            statement.generate_file_export()
+
+        statement.compute_statement()
+        file_content = statement.with_context(purchase=True).generate_file_export()
+        self.assertIn(bill_refund.partner_id.vat[2:], file_content)
+
+        # Monthly Purchase file lengths
+        # File head line: 75
+        # Frontispiece: 130
+        # Goods bill: 119
+        file_lines = file_content.splitlines()
+        self.assertEqual(len(file_lines), 3)
+        self.assertSetEqual({len(line) for line in file_lines}, {75, 96, 130})
```

Comment:
```suggestion
        # create a second bill for the same partner to check that refund is
        # not subtracted
        bill1 = self._get_intrastat_computed_bill(
            price_unit=333.00, date="2019-01-08")

        bill_refund = self._create_move_refund(bill, date="2019-01-10")
        # This refund will not be subtracted from bill
        bill_refund.update(
            {
                "intrastat": True,
            }
        )
        bill_refund.action_post()
        bill_refund.compute_intrastat_lines()

        statement = self.statement_model.create(
            {
                "period_number": bill_refund.invoice_date.month,
                "fiscalyear": bill_refund.invoice_date.year,
            }
        )

        # Check that before computation, file generation raises an exception
        # because statement is still empty
        with self.assertRaises(ValidationError):
            statement.generate_file_export()

        statement.compute_statement()
        self.assertTrue(statement.purchase_section1_ids)
        self.assertAlmostEqual(
            bill1.amount_untaxed,
            statement.purchase_section1_ids.amount_euro)
        self.assertTrue(statement.purchase_section2_ids)
        self.assertAlmostEqual(
            bill_refund.amount_untaxed,
            statement.purchase_section2_ids.amount_euro)
        file_content = statement.with_context(purchase=True).generate_file_export()
        self.assertIn(bill_refund.partner_id.vat[2:], file_content)

        # Monthly Purchase file lengths
        # File head line: 75
        # Frontispiece: 130
        # Goods bill: 119
        file_lines = file_content.splitlines()
        self.assertEqual(len(file_lines), 4)
        self.assertSetEqual({len(line) for line in file_lines}, {75, 96, 119, 130})
```

Reply:
Si ho rivisto il test e corretto l'altro sul purchase